File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int maxLen = 0;
        unordered_set<int> seen(nums.begin(), nums.end());
        for(int num : seen){
            if(!seen.count(num - 1)){
                int len = 1;
                int curr = num;

                while(seen.count(curr + 1)){
                    curr = curr + 1;
                    len++;
                }

                maxLen = max(maxLen, len);
            }
        }

        return maxLen;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        seen = set(nums)
        maxLen = 0

        for num in seen:
            if num - 1 not in seen:
                curr = num
                currLen = 1

                while curr + 1 in seen:
                    curr += 1
                    currLen += 1

                maxLen = max(maxLen, currLen)

        return maxLen
        